### PR TITLE
feat: add Move language support

### PR DESCRIPTION
**Pull request by hariria**
## Description

- [x] Adding language support for `move`, a smart contract language for several L1 blockchains.
- [x] Added a `smart contract` category for languages

## Test

`pnpm run fetch && pnpm run play`

### CODE REVIEW

**Comment by antfu** [comment]
https://github.com/move-language/move Seems to be inactive for 6 months, while https://github.com/pontem-network/vscode-move-ide is inactive for 3 years and even archived.

Is Move the language still active?

**Comment by hariria** [comment]
Sorry, I was linking to the wrong grammar reference. I updated it, this is the correct one: https://github.com/damirka/move-syntax. 

The Move language is definitely still active. It's also officially supported on GH as a language (https://github.com/github-linguist/linguist/pull/5955).

**Comment by antfu** [comment]
I am not sure about the `smart contract` category - I don't think it's general enough to be a category

**Comment by hariria** [comment]
Sure no problem!

**Comment by hariria** [comment]
@antfu lmk how it looks!

File: sources-grammars.ts [hunk 1]
```diff
@@ -648,6 +648,12 @@ export const sourcesCommunity: GrammarSource[] = [
     displayName: 'Mojo',
     source: 'https://github.com/modularml/mojo-syntax/blob/main/syntaxes/mojo.syntax.json',
   },
+  {
+    name: 'move',
+    displayName: 'Move',
+    aliases: ['mv'],
```
Comment (antfu) [inline sources-grammars.ts:654]: I don't see `mv` has been used https://github.com/github-linguist/linguist/pull/5955/files#diff-9c337ec09ccaaa1374713aad9ba26cdc1308586a27cb309a883a0b19bde663d8R4070, do you have a reference?

Comment (hariria) [reply sources-grammars.ts:654]: `.mv` is the bytecode format of the `move` source code (i.e., https://github.com/diem/move/blob/main/language/move-binary-format/src/unit_tests/empty_script.mv)

Comment (hariria) [reply sources-grammars.ts:654]: If you would prefer I not include aliases for it I can remove it?

Comment (antfu) [reply sources-grammars.ts:654]: So actually `.mv` is different from `.move`, right? In that sense I think we could not include the alias

Comment (hariria) [reply sources-grammars.ts]: You got it boss 👍

**Comment by hariria** [comment]
Cc @antfu i think it should be g2g now

**Comment by hariria** [comment]
@antfu lmk if this looks good to you now! i removed the .mv alias

File: samples/move.sample
Comment (antfu) [inline samples/move.sample:1]: This sample is way too long - can we replace it with something like https://github.com/move-language/move/blob/main/language/documentation/tutorial/step_2/BasicCoin/sources/FirstModule.move?

**Comment by antfu** [comment]
Please avoid pinging, we will get PRs reviewed in orders. I keep the pinging feature enabled for emergency use. Thanks.

**Comment by hariria** [comment]
Sorry for the pinging, thanks for the update, I've gone ahead and changed the example

**Review by antfu — APPROVED** [review]

**Pull request merged**